### PR TITLE
Add support for Raspberry Pi Pico RP2040 board 

### PR DESCRIPTION
**Pull request by cyber-murmel**
Sorry for closing and reopening this PR. I was cleaning up my fork to be forked from this repo.

---

This PR provides the files to run apollo on the Raspberry Pi Pico.

Compile with
```shell
make APOLLO_BOARD=raspberry_pi_pico
```

## Pin Map
```
	TMS_GPIO = 5,
	TDI_GPIO = 3, // MOSI
	TDO_GPIO = 4, // MISO
	TCK_GPIO = 6, // SCK
```

I also got code for other RP2040 boards, but they aren't supported yet in the upstream tinyusb.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "lib/tinyusb"]
 	path = lib/tinyusb
 	url = https://github.com/hathach/tinyusb
+[submodule "lib/pico-sdk"]
+	path = lib/pico-sdk
+	url = https://github.com/raspberrypi/pico-sdk.git
```
Comment (mndza) [inline .gitmodules:6]: Could we use the `PICO_SDK_FETCH_FROM_GIT=1` CMake variable instead of adding the pico-sdk as a submodule?

File: firmware/src/boards/raspberry_pi_pico/usb_switch.c
Comment (mndza) [inline firmware/src/boards/raspberry_pi_pico/usb_switch.c:1]: Can we avoid moving `usb_switch.c` and `button.c` to a board-defined subdirectory? I understand that the problem is the following line:
```c
#include <hal/include/hal_gpio.h>
``` 
But in fact this line can be removed for all boards, as `apollo_board.h` already includes it for the SAMDx1 MCUs.

**Review by mndza — CHANGES_REQUESTED** [review]
Thank you! I added some suggestions.

File: .gitmodules (hunk 1 above)
Comment (cyber-murmel) [reply .gitmodules]: The makefile now checks if `PICO_SDK_PATH` is defined and acts accordingly.
https://github.com/greatscottgadgets/apollo/pull/34/commits/e1f5a04bd7033b949b4217533119a6e4ee0485b7#diff-43e5a6ff56e05cff146ed6f928508217da39b0db828c180d7de01e57b42d0790R12

**Comment by cyber-murmel** [comment]
I can't reproduce the failed build on my machine.

File: firmware/src/button.c
Comment (mndza) [inline firmware/src/button.c:1]: Could you revert the changes to this file, and just remove the `#include <hal...` line?

File: firmware/src/mcu/rp2040/usb_descriptors.c
Comment (mndza) [inline firmware/src/mcu/rp2040/usb_descriptors.c:1]: I reckon this file and `dfu.c` need indentation fixes to be consistent with the rest of the code.

**Comment by cyber-murmel** [comment]
The copyright and license notice are currently still very inconsistent. Do you have a preference on what to set them to?

**Comment by mossmann** [comment]
Thank you so much for the contribution!
